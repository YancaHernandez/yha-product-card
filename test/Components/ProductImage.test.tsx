import React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components';

import { product2 } from "../data/products";
describe('ProductImage', () => {
    it('debe de mostrar el componente con la imagen  personalizada', () => {
        const wrapper = renderer.create(<ProductImage  img="https://cdn.pixabay.com/photo/2012/04/05/01/24/dice-25637_960_720.png" />);

        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    test('debe de mostrar el componente con la imagen del producto', () => {
        const wrapper = renderer.create(
            <ProductCard product={product2} >
                {
                    () => <ProductImage />
                }
            </ProductCard>
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    });
});