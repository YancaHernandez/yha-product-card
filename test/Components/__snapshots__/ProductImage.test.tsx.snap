// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage debe de mostrar el componente con la imagen  personalizada 1`] = `
<img
  alt="Product"
  className="productImg undefined"
  src="https://cdn.pixabay.com/photo/2012/04/05/01/24/dice-25637_960_720.png"
/>
`;

exports[`ProductImage debe de mostrar el componente con la imagen del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
